### PR TITLE
Add argparse/2.1 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **argparse/2.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/argparse/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses", keep_path=False)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib"))
+
+    def package_id(self):
+        del self.info.settings.compiler
```
Comment (danimtb) [inline recipes/argparse/all/conanfile.py]: not needed
```suggestion
```

**Comment by madebr** [comment]
@uilianries `tools.check_min_cppstd` checks for the *default* cpp standard of the compiler.
What I want to check against is the *minimum supported* cpp standard of the compiler

**Comment by uilianries** [comment]
> @uilianries `tools.check_min_cppstd` checks for the _default_ cpp standard of the compiler.
> What I want to check against is the _minimum supported_ cpp standard of the compiler

It checks _default_ cpp standard of the compiler IF settings.compiler.cppstd is not declared only.

**Comment by madebr** [comment]
I'm going to use `tools.valid_min_cppstd()` and print an error.
Because this is a header-only package, creating the package should always succeed.
Consumers should check the errors when installing.

**Comment by uilianries** [comment]
Please, revert 2 commits before and only update the first condition as I mentioned. The idea behind check_min_cppstd is only when cppstd is declared and configured, otherwise, you need to check the compiler version.

**Comment by madebr** [comment]
I don't want to test against the default c++ standard, I want to test against the maximum supported c++ standard of the compiler.
So users can decide for themselves what to do.

**Comment by uilianries** [comment]
Again, `check_min_cppstd` won't check the default cpp version supported by the compiler on PR madebr#9 it will ONLY check the `settings.compiler.cppstd` IF it is defined. You can run conanio/gcc7 and verify.

**Comment by madebr** [comment]
Together we're stronger :muscle:

File: recipes/argparse/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    def configure(self):
+        if self.settings.get_safe("compiler.cppstd"):
+            tools.check_valid_cppstd(self, "17")
+        try:
+            minimum_required_compiler_version = self._compiler_required_cpp17[str(self.settings.compiler)]
+            if tools.Version(self.settings.compiler.version) < minimum_required_compiler_version:
+                raise ConanInvalidConfiguration("This package requires c++17 support. The current compiler does not support it.")
+        except KeyError:
+            self.output.info("This recipe has no support for the current compiler. Please consider adding it.")
```
Comment (danimtb) [inline recipes/argparse/all/conanfile.py]:
```suggestion
            self.output.warn("This recipe has no support for the current compiler. Please consider adding it.")
```

**Pull request merged**